README.md

```diff
@@ -1,94 +1,187 @@
 
-To learn more about Juju, visit https://juju.is/docs/olm.
+## Give it a try!
 
+Let's use `ops` to build a Kubernetes charm!
 
-## Pure Python
+### Set up
 
-The framework provides a standardised Python object model that represents the
-application graph, as well as an event-handling mechanism for distributed
-system coordination and communication.
+You will need Charmcraft, a Kubernetes cloud, and Juju. The quickest way is to use a Multipass VM launched with the `charm-dev` blueprint. 
 
-The latest version of ops requires Python 3.8 or above.
+Install Multipass: [Linux](https://multipass.run/docs/installing-on-linux) | [macOS](https://multipass.run/docs/installing-on-macos) | [Windows](https://multipass.run/docs/installing-on-windows). On Linux:
 
-Juju itself is written in Go for efficient concurrency even in large
-deployments. Charms can be written in any language, however, we recommend using
-Python with this framework to make development easier and more standardised.
-All new charms at Canonical are written using it.
+```
+sudo snap install multipass
+```
+
+Use Multipass to launch an Ubuntu VM with the `charm-dev` blueprint: 
+
+```
+multipass launch --cpus 4 --memory 8G --disk 30G --name tutorial-vm charm-dev 
+```
+
+Open a shell into the VM:
+
+```
+multipass shell tutorial-vm
+```
+
+Verify that you have Charmcraft, MicroK8s, and Juju:
+
+```
+snap list # should show all three
+```
+
+Verify that Juju recognizes MicroK8s as a localhost cloud:
+
+```
+juju clouds # should show both
+```
+
+Bootstrap a Juju controller into the MicroK8s cloud:
+
+```
+juju bootstrap microk8s tutorial-controller
+```
+
+Add a workspace, or 'model':
+
+```
+juju add-model tutorial-model
+```
+
+### Write your charm
 
+Create a charm directory and use Charmcraft to initialise your charm file structure:
 
-## Getting started
+```
+mkdir my-new-charm; cd my-new-charm; charmcraft init
+```
+This has created a standard charm directory structure. Poke around.
 
-A package of operator code is called a charmed operator or simply "charm".
-You'll use [charmcraft](https://juju.is/docs/sdk/install-charmcraft) to
-register your charm name and publish it when you are ready. You can follow one
-of our [charming tutorials](https://juju.is/docs/sdk/tutorials) to get started
-writing your first charm.
+Note that the `metadata.yaml` file shows that what we have is an example charm called `my-new-charm`, which uses an OCI image resource `httpbin` from `kennethreitz/httpbin`.
 
+Note that the `requirements.txt` file lists a version of `ops`. 
 
-## Testing your charms
+Note that the `src/charm.py` file imports `ops` and uses `ops` constructs to create a charm class `MyNewCharmCharm`, observe Juju events, and pair them to event handlers:
+
+```
+...
+import ops
+...
+class MyNewCharmCharm(ops.CharmBase):
```

Comment:
```suggestion
class MyNewCharm(ops.CharmBase):
```

Reply:
That was the automatic name in the code, based on the name of the directory. I could change the name of the directory. What would make sense? `foo`? Not great...

Reply:
Perhaps we can call the directory `ops-example`, and then the charm would be called `OpsExampleCharm`? This also fits better with "No charm or operator suffix" in https://juju.is/docs/sdk/naming#heading--naming-charms

Reply:
Done.